### PR TITLE
check/uncheck all in testing tab and Analysis tab

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

**Comment by yann300** [comment]
@LianaHus  please check test errors, there are some typo there

**Comment by LianaHus** [comment]
do you want to add this into release?

**Comment by yann300** [comment]
@LianaHus yes but the typo should be fixed

**Comment by yann300** [comment]
i could not reproduce each time but it happens that if there are 2 tests files, both are executed even though only one is selected.
will try to come up with a way to reproduce

**Comment by LianaHus** [comment]
thanks, will check

File: src/app/staticanalysis/staticAnalysisView.js [hunk 1]
```diff
@@ -134,29 +134,23 @@ staticAnalysisView.prototype.run = function () {
...
-  for (var i = 0; i < all.length; i++) {
-    if (all[i].checked === true) {
-      isAnySelected = true
-      break
-    }
-  }
-  for (var j = 0; j < all.length; j++) {
-    all[j].checked = !isAnySelected
+staticAnalysisView.prototype.checkModule = function (event) {
+  let selected = document.querySelectorAll('[name="staticanalysismodule"]:checked')
```
Comment (yann300) [inline src/app/staticanalysis/staticAnalysisView.js]: seems you don't use `selected` in this function

File: src/app/staticanalysis/staticAnalysisView.js [hunk 2]
```diff
@@ -134,29 +134,23 @@ staticAnalysisView.prototype.run = function () {
...
+  let checkAll = document.querySelector('[id="checkAllEntries"]')
```
Comment (yann300) [inline src/app/staticanalysis/staticAnalysisView.js]: it's better using `this.view` than `document` for querying something

File: src/app/tabs/test-tab.js [hunk 3]
```diff
@@ -132,13 +132,32 @@ module.exports = class TestTab {
...
-      return tests.map(test => yo`<label><input onchange =${(e) => toggleCheckbox(e, test)} type="checkbox" checked="true">${test} </label>`)
+      return tests.map(test => yo`<label class="singleTestLabel"><input class="singleTest" onchange =${(e) => toggleCheckbox(e.target.checked, test)} type="checkbox" checked="true">${test} </label>`)
     }
 
-    function toggleCheckbox (e, test) {
-      var selectedTests = self.data.selectedTests
-      selectedTests = e.target.checked ? [...selectedTests, test] : selectedTests.filter(el => el !== test)
+    function toggleCheckbox (eChecked, test) {
+      if (!self.data.selectedTests) {
+        self.data.selectedTests = document.querySelectorAll('.singleTest:checked')
```
Comment (yann300) [inline src/app/tabs/test-tab.js]: better to use `this.view` here if possible

File: src/app/tabs/test-tab.js [hunk 4]
```diff
@@ -132,13 +132,32 @@ module.exports = class TestTab {
...
+      let checkAll = document.querySelector('[id="checkAllTests"]')
+      if (eChecked) {
+        checkAll.checked = true
+      } else if (!selectedTests.length) {
+        checkAll.checked = false
+      }
+    }
+
+    function checkAll (event) {
+      let checkBoxes = document.querySelectorAll('.singleTest')
```
Comment (yann300) [inline src/app/tabs/test-tab.js]: better to use `this.view` here if possible

File: src/app/tabs/test-tab.js [hunk 5]
```diff
@@ -132,13 +132,32 @@ module.exports = class TestTab {
...
+      const checkboxesLabels = document.querySelectorAll('.singleTestLabel')
+      // checks/unchecks all
+      for (let i = 0; i < checkBoxes.length; i++) {
+        checkBoxes[i].checked = event.target.checked
+        toggleCheckbox(event.target.checkedt, checkboxesLabels[i].innerText)
```
Comment (yann300) [inline src/app/tabs/test-tab.js]: seems there's a typo here: `event.target.checkedt`

**Comment by LianaHus** [comment]
@yann300  could you plz review this ones more? TIA

**Pull request merged**